micro-rdk/src/common/camera/fake_camera.rs

```diff
@@ -130,25 +111,26 @@ mod tests {
             attributes: None,
             ..Default::default()
         }));
+        let mut registry: Box<ComponentRegistry> = Box::default();
 
-        robot.process_components(conf, Box::default())?;
+        robot.process_components(conf, &mut registry)?;
 
         Ok(robot)
     }
 
     async fn setup_grpc_server(exec: Executor, addr: SocketAddr) {
-        let mut listener = NativeListener::new((addr).into(), None)
-            .unwrap()
-            .as_async_listener()
-            .await
-            .unwrap();
+        let tcp_server = TcpListener::bind(addr);
+        assert!(tcp_server.is_ok());
+        let tcp_server = tcp_server.unwrap();
+        let listener: async_io::Async<TcpListener> = tcp_server.try_into().unwrap();
 
         let robot = Arc::new(Mutex::new(setup_robot().unwrap()));
 
         loop {
             let incoming = listener.accept().await;
             assert!(incoming.is_ok());
-            let stream = incoming.unwrap();
+            let incoming = incoming.unwrap();
+            let stream: NativeStream = NativeStream::LocalPlain(incoming.0);
```

Comment:
What is the context for these changes? They seem somewhat unrelated to the rest?

Reply:
the camera tests don't pass without it, ever since [this commit](https://github.com/viamrobotics/micro-rdk/commit/b4f543140e2bb919d7294c97352f35d69b27dbed)